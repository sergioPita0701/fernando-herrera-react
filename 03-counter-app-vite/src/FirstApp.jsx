import PropTypes from "prop-types";

export const FirstApp = ({ title, subTitle,name }) => {
  return (
    <>
      <div data-testid = "test-title">{title}</div>
      <h2>{subTitle}</h2>
      <h2>{subTitle}</h2>

      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = { 
  title: "No hay Titulos",
  subTitle: "No hay Subtitulos",
 // name: 'no hay nombre'
};
