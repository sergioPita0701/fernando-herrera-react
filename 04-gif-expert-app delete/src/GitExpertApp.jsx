import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GitExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([...categories, newCategory]);
        //setCategories([categories => [...categories,'Valoran']);
    }

    return (
        <>
            <h1>GitExpertApp</h1>

            <AddCategory
                onNewCategory={(value) => onAddCategory(value)}
            />

            {
                categories.map((category) => (
                    <GifGrid key={category} category={category} />
                )
                )
            }
        </>
    )
}
