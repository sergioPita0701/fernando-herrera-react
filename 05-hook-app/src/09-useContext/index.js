export * from './AboutPage'
export * from './HomePage'
export * from './LoginPge'
export * from './MainAPP'

