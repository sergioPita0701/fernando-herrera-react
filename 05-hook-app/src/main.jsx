import React from "react";
import ReactDOM from "react-dom/client";
import {
  BrowserRouter,
  createBrowserRouter,
  RouterProvider,
} from "react-router-dom";
// import { CallBackHook } from "./06-memos/CallBackHook";
// import { MemorizeHook } from "./06-memos/MemoHook";
// import { Padre } from "./07-tarea-memo/Padre";
// import { TodoApp } from "./08-useReducer/TodoApp";
import { MainApp } from "./09-useContext/MainAPP";

// import { FocusScreen } from "./04-useRef/FocusScreen";
// import { Layout } from "./05-useLayaotEfect/Layout";
// import { Memorize } from "./06-memos/Memorize";
// import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook";
// import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
// import { CounterApp } from "./01-useState/CounterApp";
// import { CounterWhitCustomHook } from "./01-useState/CounterWhitCustomHook";
// import { SimpleForm } from "./02-useEffect/SimpleForm";
// import './08-useReducer/intro-reducer'

import "./index.css";
const router = createBrowserRouter([
  {
    path: "/",
    element: <div>Hello world!</div>,
  },
]);
ReactDOM.createRoot(document.getElementById("root")).render(
  <BrowserRouter>
    {/* // <React.StrictMode> */}
    <MainApp />
    {/* //  </React.StrictMode> */}
  </BrowserRouter>
);
