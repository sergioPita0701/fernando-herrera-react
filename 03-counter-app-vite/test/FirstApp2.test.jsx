import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"


describe('pruebas en FirstApp', () => {

    const title = 'Hola, soy Sergio';
    const subTitle = 'Soy un subtitulo';


    test('debe de hacer match con el snapshot', () => {

        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar el hol soy gouk', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    })

    test('debe de mostrar el titulo en h1', () => {

        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);

    })

    test('debe mostrar que el subtitulo enviado por porps', () => {

        render(
            <FirstApp
                title={title}
                subTitle={subTitle}
            />
        );
        expect(screen.getAllByText(subTitle).length).toBe(2)

    });



})