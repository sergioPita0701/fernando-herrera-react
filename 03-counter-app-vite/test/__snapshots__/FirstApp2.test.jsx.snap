// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en FirstApp debe de hacer match con el snapshot 1`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Sergio
  </div>
  <h2>
    No hay Subtitulos
  </h2>
  <h2>
    No hay Subtitulos
  </h2>
  <p />
</div>
`;

exports[`pruebas en FirstApp debe de hacer match con el snapshot 2`] = `
<div>
  <div
    data-testid="test-title"
  >
    Hola, soy Sergio
  </div>
  <h2>
    No hay Subtitulos
  </h2>
  <h2>
    No hay Subtitulos
  </h2>
  <p />
</div>
`;
